front/app/components/ParticipationCTABars/VotingCTABar/CTAButton.tsx

```diff
@@ -0,0 +1,140 @@
+import React, { useState } from 'react';
+
+// api
+import useBasket from 'api/baskets/useBasket';
+import useUpdateBasket from 'api/baskets/useUpdateBasket';
+import useVoting from 'api/baskets_ideas/useVoting';
+import useAppConfiguration from 'api/app_configuration/useAppConfiguration';
+
+// components
+import { Box, Text, Icon, Button } from '@citizenlab/cl2-component-library';
+import Tippy from '@tippyjs/react';
+
+// styling
+import { useTheme } from 'styled-components';
+
+// i18n
+import { FormattedMessage, useIntl } from 'utils/cl-intl';
+import messages from '../messages';
+import useLocalize from 'hooks/useLocalize';
+
+// utils
+import JSConfetti from 'js-confetti';
+import { scrollToElement } from 'utils/scroll';
+import { getDisabledExplanation } from './utils';
+
+// typings
+import { IProjectData } from 'api/projects/types';
+import { IPhaseData } from 'api/phases/types';
+
+const confetti = new JSConfetti();
+
+interface Props {
+  participationContext: IProjectData | IPhaseData;
+}
+
+const CTAButton = ({ participationContext }: Props) => {
+  const [processing, setProcessing] = useState(false);
+
+  const basketId = participationContext.relationships.user_basket?.data?.id;
+
+  const { data: basket } = useBasket(basketId);
+  const { mutate: updateBasket } = useUpdateBasket();
+  const { numberOfVotesCast, processing: votingProcessing } = useVoting();
+  const { data: appConfig } = useAppConfiguration();
+  const theme = useTheme();
+  const { formatMessage } = useIntl();
+  const localize = useLocalize();
+
+  const votingMethod = participationContext.attributes.voting_method;
+  if (!votingMethod || numberOfVotesCast === undefined) return null;
+
+  const currency = appConfig?.data.attributes.settings.core.currency;
+
+  const disabledExplanation = getDisabledExplanation(
```

Comment:
Extracted this button in a separate component, and changed a bit how the disabled explanation gets calculated